### PR TITLE
Update pynxos to 0.0.4 and add verify optional_args to nxos

### PR DESCRIPTION
**Pull request by targuan**
Update current version of pynxos >= 0.0.4: #784
Verify argument is exposed via ``verify`` key in ``optional_args``. It defaults to ``False`` for compatibility and can be set to ``True`` or any value allowed by requests library.
I kept the InsecureRequestWarning warnings from urllib3on purpose. These warnings can be easily blacked out.

I don't knwo how this code can be tested as the mocked driver use for the test bypass the open method.
I have just tested it against a live device.

### CODE REVIEW

**Comment by ktbyers** [comment]
I will try to see if I can get this merged this weekend.

**Comment by ktbyers** [comment]
@targuan Can you update the optional_arg to be `ssl_verify`? I think `verify` is just a bit too generic. Obviously this is only on the napalm side; it will still be what it is in pynxos.

It will be easier to have you update it then me (so I can just review and approve it).

Thanks.

**Comment by targuan** [comment]
Thanks for the review.
I've updated the optional_arg to ssl_verify.

File: napalm/nxos/nxos.py [hunk 1]
```diff
@@ -199,14 +199,17 @@ def __init__(self, hostname, username, password, timeout=60, optional_args=None)
         elif self.transport == 'http':
             self.port = optional_args.get('port', 80)
 
+        self.ssl_verify = optional_args.get('verify', False)
```
Comment (ktbyers) [inline napalm/nxos/nxos.py]: You didn't change the second half of this i.e. optional_args.get is still looking for verify.

**Comment by targuan** [comment]
Sorry for the time wasted.
I did correctly tested it this time:
It raise SSLError when the check is enabled
```
(.venv) [root@localhost napalm]# napalm --user admin --pass Cisco123 --vendor nxos --optional_args 'ssl_verify=True' 10.10.20.58 call get_facts
2018-10-14 00:33:08,456 - napalm - ERROR - open - Failed: HTTPSConnectionPool(host='10.10.20.58', port=443): Max retries exceeded with url: /ins (Caused by SSLError(SSLError(1, u'[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:579)'),))
```
It complete when it ssl_verify is disabled:
```
(.venv) [root@localhost napalm]# napalm --user admin --pass Cisco123 --vendor nxos --optional_args 'ssl_verify=False' 10.10.20.58 call get_facts &> /dev/null 
(.venv) [root@localhost napalm]# echo $?
0
```

**Pull request merged**